File: 9. Palindrome Number/palindrome_number.js
function isPalindrome(x) {
    if (x < 0 || (x % 10 === 0 && x !== 0)) return false;

    let reverted = 0;
    while (x > reverted) {
        const digit = x % 10;
        reverted = reverted * 10 + digit;
        x = Math.floor(x / 10);
    }

    return x === reverted || x === Math.floor(reverted / 10);
}

function main() {
    console.log(isPalindrome(121));   // true
    console.log(isPalindrome(-121));  // false
    console.log(isPalindrome(10));    // false
    console.log(isPalindrome(0));     // true
}

main();

File: 009. Palindrome Number/palindrome_number.py
def isPalindrome(x: int) -> bool:
    if x < 0 or (x % 10 == 0 and x != 0):
        return False

    reverted = 0
    while x > reverted:
        digit = x % 10
        reverted = reverted * 10 + digit
        x //= 10

    return x == reverted or x == reverted // 10

if __name__ == "__main__":
    print(isPalindrome(121))   # True
    print(isPalindrome(-121))  # False
    print(isPalindrome(10))    # False
    print(isPalindrome(0))     # True
